File: code/python/disk_partition.py
import logging
from utils import run_command

logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")

def partition_disk(disk):
    logging.info(f"Partitioning disk {disk}...")

    try:
        run_command(["parted", f"/dev/{disk}", "--script", "mklabel", "gpt"])
        run_command(["parted", f"/dev/{disk}", "--script", "mkpart", "primary", "0%", "100%"])
        logging.info(f"Disk {disk} partitioned successfully.")
    except FileNotFoundError:
        logging.info(f"Error: `parted` command not found. Ensure it is installed.")
    except subprocess.CalledProcessError:
        logging.info(f"Error: Failed to partition {disk}.")

File: code/c/src/disk_partition.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "disk_partition.h"
#include "utils.h"

void partition_disk(const char *disk) {
    printf("Partitioning disk %s...\n", disk);

    char cmd[256];
    snprintf(cmd, sizeof(cmd), "parted /dev/%s --script mklabel gpt", disk);
    if (system(cmd) != 0) {
        perror("Error creating partition label");
        exit(1);
    }

    snprintf(cmd, sizeof(cmd), "parted /dev/%s --script mkpart primary 0%% 100%%", disk);
    if (system(cmd) != 0) {
        perror("Error creating partition");
        exit(1);
    }

    printf("Disk %s partitioned successfully.\n", disk);
}
